### DISCUSSION TITLE
Terminating await and restarting

### DISCUSSION THREAD

**Discussion by adrian141b** [Rapptz/discord.py#8165] [General]
hello, i would like to create a bot which Terminating await and restarting, it it possible? (i have loop error after the second bot.run("token"))

For expample:

@client.event
async def on_ready():
   downloading messages...
   print(messages saved)

x =5 
while x > 0
   bot.run("token")
   x-=1

I know we have a discord command but i can't use timing session in my project.

**Comment by rtk-rnjn** [comment]
So you are basically want a loop of which run 5 times! Better use [`ext.tasks`](https://discordpy.readthedocs.io/en/latest/ext/tasks/index.html)
```py
from discord.ext import tasks

RUN_AFTER = 60  # seconds
COUNT = 5  # 5 loops

bot = commands.Bot(...)

@tasks.loop(seconds=RUN_AFTER, count=COUNT)
async def message_downloader():
    ...

message_downloader.start()
bot.run(...)
```

**Reply by adrian141b** [reply]
sorry, but i can't use Task in my application

**Reply by adrian141b** [reply]
Do you have any other ideas?
Requirements are:
{
exit an asynchronous function 
close the session
}
restart x times without errors

**Comment by Rapptz** [comment]
Restarting the instance like this doesn't work with discord.py due to the sheer amount of resources at play. Discord also doesn't like wasted IDENTIFYs like this so it's not going to be wholly supported.

Your best bet is to redesign your problem so you can use a task like posted elsewhere.

**Reply by adrian141b** [reply]
Thank you, or writing your own library, thanks for the answer
